File: example/src/main/java/com/vmloft/develop/library/example/demo/dialog/FloatMenuActivity.java
package com.vmloft.develop.library.example.demo.dialog;

import android.view.MotionEvent;
import android.view.View;
import android.widget.Button;

import com.vmloft.develop.library.example.R;
import com.vmloft.develop.library.example.common.AppActivity;
import com.vmloft.develop.library.tools.utils.VMLog;
import com.vmloft.develop.library.tools.widget.VMFloatMenu;
import com.vmloft.develop.library.tools.widget.toast.VMToast;

import butterknife.BindView;
import butterknife.OnClick;
import butterknife.OnTouch;

/**
 * Created by lzan13 on 2017/6/20.
 *
 * 测试悬浮菜单
 */
public class FloatMenuActivity extends AppActivity {

    @BindView(R.id.float_menu_left_top_btn) Button leftTopBtn;
    @BindView(R.id.float_menu_left_bottom_btn) Button leftBottomBtn;
    @BindView(R.id.float_menu_center_btn) Button centerBtn;
    @BindView(R.id.float_menu_right_top_btn) Button rightTopBtn;
    @BindView(R.id.float_menu_right_bottom_btn) Button rightBottomBtn;

    private int touchX;
    private int touchY;
    private VMFloatMenu mFloatMenu;

    @Override
    protected int layoutId() {
        return R.layout.activity_float_menu;
    }

    @Override
    protected void initUI() {
        super.initUI();
        mFloatMenu = new VMFloatMenu(mActivity);
        mFloatMenu.setItemClickListener(id -> {
            VMLog.d("点击了悬浮菜单 %d", id);
            VMToast.make(mActivity, "点击了悬浮菜单 " + id).done();
        });
    }

    @Override
    protected void initData() {

    }

    @OnTouch({
        R.id.float_menu_left_top_btn, R.id.float_menu_right_top_btn, R.id.float_menu_center_btn, R.id.float_menu_left_bottom_btn,
        R.id.float_menu_right_bottom_btn
    })
    public boolean onTouch(View v, MotionEvent event) {
        if (event.getAction() == MotionEvent.ACTION_DOWN) {
            touchX = (int) event.getRawX();
            touchY = (int) event.getRawY();
        }
        return false;
    }

    @OnClick({
        R.id.float_menu_left_top_btn, R.id.float_menu_right_top_btn, R.id.float_menu_center_btn, R.id.float_menu_left_bottom_btn,
        R.id.float_menu_right_bottom_btn
    })
    void onClick(View view) {
        switch (view.getId()) {
        case R.id.float_menu_left_top_btn:
        case R.id.float_menu_left_bottom_btn:
        case R.id.float_menu_center_btn:
        case R.id.float_menu_right_top_btn:
        case R.id.float_menu_right_bottom_btn:
            showFloatMenu(view);
            break;
        }
    }

    /**
     * 测试弹出 PopupWindow 菜单
     */
    private void showFloatMenu(View view) {
        mFloatMenu.clearAllItem();
        mFloatMenu.addItem(new VMFloatMenu.ItemBean(1, "悬浮菜单"));
        mFloatMenu.addItem(new VMFloatMenu.ItemBean(2, "悬浮"));
        mFloatMenu.addItem(new VMFloatMenu.ItemBean(3, "悬浮菜单", R.color.vm_red));
        mFloatMenu.addItem(new VMFloatMenu.ItemBean(4, "悬浮菜单菜单"));
        mFloatMenu.addItem(new VMFloatMenu.ItemBean(5, "悬浮菜单"));
        mFloatMenu.addItem(new VMFloatMenu.ItemBean(6, "菜单"));
        mFloatMenu.addItem(new VMFloatMenu.ItemBean(7, "悬浮菜单"));
        mFloatMenu.addItem(new VMFloatMenu.ItemBean(8, "浮菜单"));
        mFloatMenu.showAtLocation(view, touchX, touchY);
    }
}

File: example/src/main/java/com/vmloft/develop/library/example/demo/dialog/FloatMenuActivity.kt
package com.vmloft.develop.library.example.demo.dialog

import android.view.MotionEvent
import android.view.View
import com.alibaba.android.arouter.facade.annotation.Route
import com.vmloft.develop.library.example.R

import com.vmloft.develop.library.example.base.BaseActivity
import com.vmloft.develop.library.tools.utils.logger.VMLog
import com.vmloft.develop.library.tools.widget.VMFloatMenu
import com.vmloft.develop.library.tools.widget.VMFloatMenu.IItemClickListener
import com.vmloft.develop.library.tools.widget.VMFloatMenu.ItemBean
import com.vmloft.develop.library.tools.widget.toast.VMToast
import kotlinx.android.synthetic.main.activity_float_menu.*

/**
 * Created by lzan13 on 2017/6/20.
 *
 * 测试悬浮菜单
 */
@Route(path = "/VMLoft/FloatMenu")
class FloatMenuActivity : BaseActivity() {

    private var touchX = 0
    private var touchY = 0
    private lateinit var floatMenu: VMFloatMenu

    @Override
    override fun layoutId(): Int {
        return R.layout.activity_float_menu
    }

    @Override
    override fun initUI() {
        super.initUI()
        setTopTitle("自定义悬浮菜单")

        floatMenu = VMFloatMenu(mActivity!!)
        floatMenu.setItemClickListener(object : IItemClickListener() {
            override fun onItemClick(id: Int) {
                VMLog.d("点击了悬浮菜单 $id")
                VMToast.make(mActivity, "点击了悬浮菜单 $id").done()
            }
        })

        leftTopBtn.setOnTouchListener { view, event -> onTouch(view, event) }
        leftBottomBtn.setOnTouchListener { view, event -> onTouch(view, event) }
        centerUpBtn.setOnTouchListener { view, event -> onTouch(view, event) }
        centerDownBtn.setOnTouchListener { view, event -> onTouch(view, event) }
        rightTopBtn.setOnTouchListener { view, event -> onTouch(view, event) }
        rightBottomBtn.setOnTouchListener { view, event -> onTouch(view, event) }

        leftTopBtn.setOnClickListener { view -> onClick(view) }
        leftBottomBtn.setOnClickListener { view -> onClick(view) }
        centerUpBtn.setOnClickListener { view -> onClick(view) }
        centerDownBtn.setOnClickListener { view -> onClick(view) }
        rightTopBtn.setOnClickListener { view -> onClick(view) }
        rightBottomBtn.setOnClickListener { view -> onClick(view) }
    }

    @Override
    override fun initData() {
    }

    private fun onTouch(v: View, event: MotionEvent): Boolean {
        if (event.action == MotionEvent.ACTION_DOWN) {
            touchX = event.rawX.toInt()
            touchY = event.rawY.toInt()
        }
        return false
    }

    private fun onClick(view: View) {
        when (view.id) {
            R.id.leftTopBtn, R.id.leftBottomBtn, R.id.centerUpBtn, R.id.centerDownBtn, R.id.rightTopBtn, R.id.rightBottomBtn -> showFloatMenu(view)
        }
    }

    /**
     * 测试弹出 PopupWindow 菜单
     */
    private fun showFloatMenu(view: View) {
        floatMenu.clearAllItem()
        floatMenu.addItem(ItemBean(1, "悬浮菜单"))
        floatMenu.addItem(ItemBean(2, "悬浮"))
        floatMenu.addItem(ItemBean(3, "悬浮菜单", R.color.vm_red))
        floatMenu.addItem(ItemBean(4, "悬浮菜单", R.color.vm_accent, R.drawable.ic_close))
        floatMenu.addItem(ItemBean(5, "悬浮菜单"))
        floatMenu.addItem(ItemBean(6, "菜单"))
        floatMenu.addItem(ItemBean(7, "悬浮菜单"))
        floatMenu.addItem(ItemBean(8, "浮菜单"))
        floatMenu.setMenuBackground(R.drawable.shape_card_bg_white)
        floatMenu.showAtLocation(view, touchX, touchY)
    }
}